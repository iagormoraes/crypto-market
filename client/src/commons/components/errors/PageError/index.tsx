export default function PageError({ error }: { error?: string | string[] }) {
  if (!error) return null;

  // Get the first error on the list of errors when array
  const errorLabel = Array.isArray(error) ? error[0] : error;

  const errors: Record<string, string> = {
    'Not Found': `Sorry, we couldn't find the requested resource, try again.`,
  };

  if (!errors[errorLabel]) return null;

  return (
    <p className="my-4 text-red-500 font-bold text-center">
      {errors[errorLabel]}
    </p>
  );
}
