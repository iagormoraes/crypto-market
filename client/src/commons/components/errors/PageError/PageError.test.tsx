import { render, screen } from '@testing-library/react';
import PageError from './index';

describe('<PageError/>', () => {
  it('matches snapshot', () => {
    const { container } = render(<PageError error="Not Found" />);

    expect(container).toMatchSnapshot();
  });

  it('renders error message Not Found', () => {
    render(<PageError error="Not Found" />);

    expect(
      screen.getByText(
        `Sorry, we couldn't find the requested resource, try again.`,
      ),
    ).toBeTruthy();
  });
});
