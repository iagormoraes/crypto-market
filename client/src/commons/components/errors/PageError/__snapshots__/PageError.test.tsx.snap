// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageError/> matches snapshot 1`] = `
<div>
  <p
    class="my-4 text-red-500 font-bold text-center"
  >
    Sorry, we couldn't find the requested resource, try again.
  </p>
</div>
`;
